src/git/zcl_abapgit_git_porcelain.clas.abap

```diff
@@ -383,17 +381,26 @@ CLASS ZCL_ABAPGIT_GIT_PORCELAIN IMPLEMENTATION.
 
   METHOD full_tree.
 
-    DATA: ls_object LIKE LINE OF it_objects,
+    DATA: lv_sha1   TYPE zif_abapgit_definitions=>ty_sha1,
+          ls_object LIKE LINE OF it_objects,
           ls_commit TYPE zcl_abapgit_git_pack=>ty_commit.
 
+    IF    iv_commit IS SUPPLIED
+      AND iv_commit IS NOT INITIAL.
+      lv_sha1 = iv_commit.
+    ELSE.
+      lv_sha1 = iv_branch.
+    ENDIF.
```

Comment:
so iv_commit and iv_branch does the same thing

Reply:
s. comment above